### PR TITLE
Added a basic MO test

### LINKED ISSUE #114: Basic Media Overlay testing
I will do a basic MO test that may serve as a starting point for additional tests. Avoiding a mega PR I will do one with a single test to agree on the 'styles' (naming, file patterns, etc).

Cc: @marisademeglio I used your MO tests on the idpf test suite, to extract an individual, smaller test.

### PR DESCRIPTION
**Pull request by iherman**
Just does the basic: a mobydick audio is read over a text.

The test actually tests all statements within the [timing and sync. paragraph](https://w3c.github.io/epub-specs/epub33/rs/#sec-rsconf-timing-synch): it seemed to be an overkill in granularity to test each statements separately.

Fix #114

### CODE REVIEW

File: tests/mo-timing-synchronization/OPS/content_001.xhtml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<html xmlns="http://www.w3.org/1999/xhtml">
+	<head>
+        <meta charset="utf-8" />
+		<title>Basic Media Overlay Test</title>
+        <!-- <link rel="stylesheet" type="text/css" href="../css/base.css" /> -->
```
Comment (dlazin) [inline tests/mo-timing-synchronization/OPS/content_001.xhtml:6]: Delete this?

File: tests/mo-timing-synchronization/OPS/content_001.xhtml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	</head>
+	<body>
+
+        <p>Test passes (i.e., the Reading System implements Media Overlays) if</p>
+
+        <ol>
+            <li>The Reading System provides the user with a way to turn text reading on</li>
+            <li>If turned out, the extract on the next page is read aloud by the Reading System</li>
```
Comment (dlazin) [inline tests/mo-timing-synchronization/OPS/content_001.xhtml:14]: Turned on? Turned off? "Out" seems wrong.

File: tests/mo-timing-synchronization/OPS/mobydick.xhtml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<html xmlns="http://www.w3.org/1999/xhtml">
+	<head>
+        <meta charset="utf-8" />
+		<title>Basic Media Overlay Test</title>
+        <!-- <link rel="stylesheet" type="text/css" href="../css/base.css" /> -->
```
Comment (dlazin) [inline tests/mo-timing-synchronization/OPS/mobydick.xhtml:6]: Delete?

File: tests/mo-timing-synchronization/OPS/content_001.xhtml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<html xmlns="http://www.w3.org/1999/xhtml">
+	<head>
+        <meta charset="utf-8" />
+		<title>Basic Media Overlay Test</title>
```
Comment (dlazin) [inline tests/mo-timing-synchronization/OPS/content_001.xhtml:5]: Nit: sentence case for titles (in our tests). Applies throughout.

File: tests/mo-timing-synchronization/OPS/package.opf [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<package xmlns="http://www.idpf.org/2007/opf" xmlns:epub="http://www.idpf.org/2007/ops" version="3.0" xml:lang="en" unique-identifier="pub-id">
+  <metadata xmlns:dc="http://purl.org/dc/elements/1.1/">
+    <dc:coverage>Media Overlays</dc:coverage>
+    <dc:creator>Ivan Herman</dc:creator>
+    <dc:creator>Marisa DeMeglio</dc:creator>
+    <dc:date>2022-02-10</dc:date>
+    <dc:description>Tests whether the Reading System implements Media Overlays.</dc:description>
```
Comment (dlazin) [inline tests/mo-timing-synchronization/OPS/package.opf:8]:
```suggestion
    <dc:description>Tests whether the reading system implements media overlays.</dc:description>
```

File: tests/mo-timing-synchronization/OPS/package.opf [hunk 6]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<package xmlns="http://www.idpf.org/2007/opf" xmlns:epub="http://www.idpf.org/2007/ops" version="3.0" xml:lang="en" unique-identifier="pub-id">
+  <metadata xmlns:dc="http://purl.org/dc/elements/1.1/">
+    <dc:coverage>Media Overlays</dc:coverage>
+    <dc:creator>Ivan Herman</dc:creator>
+    <dc:creator>Marisa DeMeglio</dc:creator>
```
Comment (dlazin) [inline tests/mo-timing-synchronization/OPS/package.opf:6]: This is Marisa's first test. Will your script extract her (as a credited test author) in the second position?

File: tests/mo-timing-synchronization/OPS/package.opf [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    <dc:date>2022-02-10</dc:date>
+    <dc:description>Tests whether the Reading System implements Media Overlays.</dc:description>
+    <dc:identifier id="pub-id">mo-timing-synchronization</dc:identifier>
```
Comment (dlazin) [inline tests/mo-timing-synchronization/OPS/package.opf:9]: For consistency with other tests, consider making the abbreviation mol- instead of mo-.

**Review by dlazin — APPROVED** [review]

File: tests/mo-timing-synchronization/OPS/content_001.xhtml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+        <!-- <link rel="stylesheet" type="text/css" href="../css/base.css" /> -->
+	</head>
+	<body>
+
+        <p>Test passes (i.e., the Reading System implements Media Overlays) if</p>
+
+        <ol>
+            <li>The Reading System provides the user with a way to turn text reading on</li>
+            <li>If turned out, the extract on the next page is read aloud by the Reading System</li>
+            <li>The text is highlighed when its text is read; at first sentence by sentence, then paragraphs by paragraphs.</li>
```
Comment (marisademeglio) [inline tests/mo-timing-synchronization/OPS/content_001.xhtml:15]: How about "The text is highlighted as its audio is played", without mentioning granularity (which might get confused with granularity *switching*, e.g. any given portion can be played word by word or paragraph by paragraph, seen only in laboratory experiments thus far but often discussed as a potential feature of MO).

File: tests/mo-timing-synchronization/OPS/package.opf (hunk 5 above)
Comment (iherman) [reply tests/mo-timing-synchronization/OPS/package.opf:8]: Yep. Will come in next commit.

File: tests/mo-timing-synchronization/OPS/package.opf (hunk 7 above)
Comment (iherman) [reply tests/mo-timing-synchronization/OPS/package.opf:9]: Yep, will come in next commit

File: tests/mo-timing-synchronization/OPS/package.opf (hunk 6 above)
Comment (iherman) [reply tests/mo-timing-synchronization/OPS/package.opf:6]: No. The list of test authors, appearing at the end of the text, is ordered alphabetically.

File: tests/mo-timing-synchronization/OPS/mobydick.xhtml (hunk 3 above)
Comment (iherman) [reply tests/mo-timing-synchronization/OPS/mobydick.xhtml:6]: Yep, this is a leftover. Will do.

File: tests/mo-timing-synchronization/OPS/content_001.xhtml (hunk 1 above)
Comment (iherman) [reply tests/mo-timing-synchronization/OPS/content_001.xhtml:6]: Yep, leftover....

File: tests/mo-timing-synchronization/OPS/content_001.xhtml (hunk 4 above)
Comment (iherman) [reply tests/mo-timing-synchronization/OPS/content_001.xhtml:5]: +1

File: tests/mo-timing-synchronization/OPS/content_001.xhtml (hunk 2 above)
Comment (iherman) [reply tests/mo-timing-synchronization/OPS/content_001.xhtml:14]: +1. Will do

File: tests/mo-timing-synchronization/OPS/content_001.xhtml (hunk 8 above)
Comment (iherman) [reply tests/mo-timing-synchronization/OPS/content_001.xhtml:15]: Yes, will do

**Comment by marisademeglio** [comment]
The test talks about highlighting text but there's no associated stylesheet in which highlight styles would be defined, nor are there class definitions in the package document.

I am trying to think of what we could do instead. The issue comes down to that the statements in [Timing and Synchronization](https://w3c.github.io/epub-specs/epub33/rs/#sec-rsconf-timing-synch) are, for example, "the RS MUST render elements..." but when you look at the specific element rendering instructions, it's all SHOULDs, e.g. "SHOULD ensure the text element is visible", "SHOULD apply defined classnames". So can you evaluate a MUST assertion with behaviors that are SHOULDs?  Is this a flaw in the spec?

**Comment by iherman** [comment]
> The test talks about highlighting text but there's no associated stylesheet in which highlight styles would be defined, nor are there class definitions in the package document.
> 
> I am trying to think of what we could do instead. The issue comes down to that the statements in [Timing and Synchronization](https://w3c.github.io/epub-specs/epub33/rs/#sec-rsconf-timing-synch) are, for example, "the RS MUST render elements..." but when you look at the specific element rendering instructions, it's all SHOULDs, e.g. "SHOULD ensure the text element is visible", "SHOULD apply defined classnames". So can you evaluate a MUST assertion with behaviors that are SHOULDs? Is this a flaw in the spec?

I agree the test description may have to be more precise, but the spec seems to be o.k. to me. Rendering an audio element is defined via a MUST, and that is tested here. (Which means that the test in this PR should also be linked to that must statement. T.b.d.) What is a 'should' is the way the RS displays the text while playing the audio.

I intend to repeat this test with an associated style sheet, b.t.w. to make that part separately tested.

Note that, at least in my reading, implementation of MO is a _SHOULD_ in a first place. @dlazin we have this situation whereby we have a bunch of MUST statements but, in the final test suite, these tests should all be denoted as 'should' tests.

I will do these changes an would like to merge this PR to move ahead. We can come back to the details later, of course, but this PR just intended to see whether the overall direction is fine. (E.g., naming schemes.)

**Comment by iherman** [comment]
The test description now says:

>         <ol>
>            <li>The Reading System provides the user with a way to turn text reading on</li>
>           <li>If turned on, the extract on the next page is read aloud by the Reading System</li>
>            <li>The text may be highlighted as audio is played.</li>
>        </ol>

I think that, for the purposes of the test, this is fine. The most important change was to make the last bullet item as a 'may be'

**Pull request merged**